types/pg/index.d.ts

```diff
@@ -156,12 +161,14 @@ export class Pool extends events.EventEmitter {
     end(callback: () => void): void;
 
     query<T extends Submittable>(queryStream: T): T;
-    query(queryConfig: QueryArrayConfig, values?: any[]): Promise<QueryArrayResult>;
-    query(queryConfig: QueryConfig): Promise<QueryResult>;
-    query(queryTextOrConfig: string | QueryConfig, values?: any[]): Promise<QueryResult>;
-    query(queryConfig: QueryArrayConfig, callback: (err: Error, result: QueryArrayResult) => void): void;
-    query(queryTextOrConfig: string | QueryConfig, callback: (err: Error, result: QueryResult) => void): void;
-    query(queryText: string, values: any[], callback: (err: Error, result: QueryResult) => void): void;
+    // tslint:disable:no-unnecessary-generics
+    query<R = any[], I extends any[] = any[]>(queryConfig: QueryArrayConfig<I>, values?: I): Promise<QueryArrayResult<R>>;
```

Comment:
Note that once `QueryArrayResult` enforces `R` to extend `any[]`, `query()` will have to declare the same restriction. Note as well that this comment applies to 4 cases in total, 2 in the context of `Pool` and 2 in the context of `Client`.

Reply:
I've definitely fixed that before, no idea what happened to it. Maybe git is just being weird again...